localization/ar_tag_based_localizer/src/ar_tag_based_localizer_core.cpp

```diff
@@ -0,0 +1,276 @@
+{
+  if ((image_pub_.getNumSubscribers() == 0) && (pose_pub_->get_subscription_count() == 0)) {
+    RCLCPP_DEBUG(this->get_logger(), "No subscribers, not looking for ArUco markers");
+    return;
+  }
+
+  if (!cam_info_received_) {
+    RCLCPP_DEBUG(this->get_logger(), "No cam_info has been received.");
+    return;
+  }
+
+  builtin_interfaces::msg::Time curr_stamp = msg->header.stamp;
+  cv_bridge::CvImagePtr cv_ptr;
+  try {
+    cv_ptr = cv_bridge::toCvCopy(*msg, sensor_msgs::image_encodings::RGB8);
+  } catch (cv_bridge::Exception & e) {
+    RCLCPP_ERROR(this->get_logger(), "cv_bridge exception: %s", e.what());
+    return;
+  }
+
+  cv::Mat in_image = cv_ptr->image;
+
+  // detection results will go into "markers"
+  std::vector<aruco::Marker> markers;
+
+  // ok, let's detect
+  detector_.detect(in_image, markers, cam_param_, marker_size_, false);
+
+  // for each marker, draw info and its boundaries in the image
+  for (const aruco::Marker & marker : markers) {
+    tf2::Transform tf_cam_to_marker = right_to_left_ * aruco_marker_to_tf2(marker);
+
+    geometry_msgs::msg::TransformStamped tf_cam_to_marker_stamped;
+    tf2::toMsg(tf_cam_to_marker, tf_cam_to_marker_stamped.transform);
+    tf_cam_to_marker_stamped.header.stamp = curr_stamp;
+    tf_cam_to_marker_stamped.header.frame_id = camera_frame_;
+    tf_cam_to_marker_stamped.child_frame_id = "detected_marker_" + std::to_string(marker.id);
+    tf_broadcaster_->sendTransform(tf_cam_to_marker_stamped);
+
+    geometry_msgs::msg::PoseStamped pose_cam_to_marker;
+    tf2::toMsg(tf_cam_to_marker, pose_cam_to_marker.pose);
+    pose_cam_to_marker.header.stamp = curr_stamp;
+    pose_cam_to_marker.header.frame_id = std::to_string(marker.id);
+    publish_pose_as_base_link(pose_cam_to_marker);
+
+    // drawing the detected markers
+    marker.draw(in_image, cv::Scalar(0, 0, 255), 2);
+  }
+
+  // draw a 3d cube in each marker if there is 3d info
+  if (cam_param_.isValid() && marker_size_ != -1) {
+    for (aruco::Marker & marker : markers) {
+      aruco::CvDrawingUtils::draw3dAxis(in_image, marker, cam_param_);
+    }
+  }
+
+  if (image_pub_.getNumSubscribers() > 0) {
+    // show input with augmented information
+    cv_bridge::CvImage out_msg;
+    out_msg.header.stamp = curr_stamp;
+    out_msg.encoding = sensor_msgs::image_encodings::RGB8;
+    out_msg.image = in_image;
+    image_pub_.publish(out_msg.toImageMsg());
+  }
+}
+
+// wait for one camera info, then shut down that subscriber
+void ArTagBasedLocalizer::cam_info_callback(const sensor_msgs::msg::CameraInfo & msg)
+{
+  if (cam_info_received_) {
+    return;
+  }
+
+  cam_param_ = ros_camera_info_to_aruco_cam_params(msg, true);
+
+  // handle cartesian offset between stereo pairs
+  // see the sensor_msgs/CameraInfo documentation for details
+  right_to_left_.setIdentity();
+  right_to_left_.setOrigin(tf2::Vector3(-msg.p[3] / msg.p[0], -msg.p[7] / msg.p[5], 0.0));
```

Comment:
Sorry, this part was copied and pasted from aruco_ros, and this was a meaningless variable for ar_tag_based_localizer.
This is because `p` in `CameraInfo` is Projection/camera matrix, and the 3rd and 7th of this are translation values, but in the data currently used, these are 0, `right_to_left_` is always an Identity Matrix.

It is not even necessary from an view of algorithm. 

Fixed in https://github.com/autowarefoundation/autoware.universe/pull/4347/commits/f2f9a60b9e53ed466784fa2240b5f0389e37be4d